src/ast/mod.rs

```diff
@@ -812,6 +815,71 @@ impl FromStr for FileFormat {
     }
 }
 
+/// A `LISTAGG` invocation
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct ListAgg {
+    pub distinct: bool,
+    pub expr: Box<Expr>,
+    pub separator: Option<Box<Expr>>,
+    pub on_overflow: Option<ListAggOnOverflow>,
+    pub within_group: Option<Vec<OrderByExpr>>,
+}
+
+impl fmt::Display for ListAgg {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let distinct = if self.distinct { "DISTINCT " } else { "ALL " };
+        let args = if let Some(separator) = &self.separator {
+            format!(
+                "{}",
+                display_comma_separated(&[self.expr.clone(), separator.clone()])
```

Comment:
It's less complex: both contenders already use `format!()`, but what's expressed as an extra 4 characters `", {}"` in one case, requires an additional `display_comma_separated()`call, two `.clone()` calls, `&[]` slice construction, which rustfmt spreads across 4 lines...

`display_comma_separated()` was introduced to avoid boilerplate working with sequences of unrestricted length, and we don't use for a fixed-length sequences elsewhere.